src/pf/existing/detect_base_dir.cpp

```diff
@@ -0,0 +1,53 @@
+#include "./detect_base_dir.hpp"
+
+#include <algorithm>
+#include <cstdlib>  // std::abort
+#include <fstream>
+#include <string>
+#include <utility>
+
+#include <boost/range/iterator_range.hpp>
+
+namespace fs = pf::fs;
+
+namespace {
+std::optional<fs::path> parse_cmakecache_homedir(fs::path const& cmakecache) {
+    std::string line;
+
+    std::ifstream file{cmakecache};
+
+    while (std::getline(file, line)) {
+        // if (line.starts_with("CMAKE_HOME_DIRECTORY"))
+        if (line.rfind("CMAKE_HOME_DIRECTORY", 0) == 0) {
```

Comment:
Commented line. Also, may want to find `CMAKE_HOME_DIRECTORY:` (with a trailing colon) to be sure to not find a false `CMAKE_HOME_DIRECTORY_FOO`

Reply:
It's not actually a commented line of code, it was intended to explain what the following line does. `line.rfind(...)` is not obviously the same as `line.starts_with(...)`. I can turn that comment into a worded explanation if necessary